tests/ballerina-compiler-api-test/src/test/resources/test-src/expressions_test.bal

```diff
@@ -358,3 +358,10 @@ function fooFn() returns future<int> {
 function barFn() returns int {
     return 10;
 }
+
+function testRegexp() {
+    _ = re `[a-z0-9]+`;
+    _ = re ``;
+    _ = re `a`;
+    _ = re `[a|b]+c`;
```

Comment:
Add two tests with expected type
regexp:RegExp and string:RegExp

Reply:
Added in https://github.com/ballerina-platform/ballerina-lang/pull/39466/commits/69edd7f4444b7efee9a2c0ea079479f2f0ca18b3.